plugin/helpers.go

```diff
@@ -69,3 +78,29 @@ func (p *HelpersImpl) ensureServerVersion(required string) error {
 	}
 	return nil
 }
+
+func (p *HelpersImpl) InstallPluginFromUrl(downloadUrl string, replace bool) (*model.Manifest, error) {
+	parsedUrl, _ := url.Parse(downloadUrl)
+	if !*p.API.GetConfig().PluginSettings.AllowInsecureDownloadUrl && parsedUrl.Scheme != "https" {
+		return nil, errors.New("downloading from insecure url is not allowed")
+	}
+
+	client := &http.Client{Timeout: 60 * time.Minute}
+	response, err := client.Get(downloadUrl)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to download the plugin")
+	}
+	defer response.Body.Close()
+
+	fileBytes, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to read response")
+	}
+
+	manifest, appError := p.API.InstallPlugin(bytes.NewReader(fileBytes), true)
```

Comment:
```suggestion
	manifest, appError := p.API.InstallPlugin(response.Body, true)
```
Hey Rajat, can you directly pass `response.Body` to `InstallPlugin()` instead of creating a byte reader?

We can also delete the line about `ioutil.ReadAll()` now, since we already will be passing an `io.Reader` to `InstallPlugin()` and it'll do the reading inside.

Reply:
Nice catch. Made changes.